specs/PointOfInterest/PointOfInterest/schema.json

```diff
@@ -37,11 +37,11 @@
       }
     }
   ],
+  "anyOf": [{"required": ["location"]}, {"required": ["address"]}],
```

Comment:
I think you need something like

```
"allOf": [
{
"anyOf": [{"required": ["location"]}, {"required": ["address"]}],
}
{
"required": [
    "id",
    "type",
    "category",
    "name"
  ]
}]
```
please check

Reply:
I don't know if there is any semantical difference. Also the `required` field is currently at the root level and not nested inside "allOf". AFAIK both are valid json-schemas and indeed both validate correctly (with ajv),  the following example object:
```
    {
        "id": "urn:ngsi-ld:PointOfInterest:Milan:Pharmacies:GeoJson:ds85-1",
        "type": "PointOfInterest",
        "address": {
            "addressLocality": "Milano",
            "addressRegion": "IT",
            "streetAddress": "WASHINGTON GIORGIO  VIA , 74"
        },
        "category": [
            "80"
        ],
        "description": "FARMACIE COMUNALI IN ESERCIZIO, PropertyType:COMUNE",
        "name": "COMUNALE N.75",
        "source": "http://dati.comune.milano.it/dataset/ds85_infogeo_farmacie_localizzazione_"
    }
```
Looking at other Data Model schemas, we have:
- [TrafficFlowObserved](https://github.com/Fiware/dataModels/blob/master/specs/Transportation/TrafficFlowObserved/schema.json), [CrowdFlowObserved](https://github.com/Fiware/dataModels/blob/master/specs/Transportation/CrowdFlowObserved/schema.json) and [EVChargingStation](https://github.com/Fiware/dataModels/blob/master/specs/Transportation/EVChargingStation/schema.json) have `allOf`, `required` and `anyOf` at the same root level. 
- [KeyPerformanceIndicator](https://github.com/Fiware/dataModels/blob/master/specs/KeyPerformanceIndicator/schema.json) has a more nested/tricky structure:  
```
"allOf": [
   .....
{
      "allOf": [
        {
          "required": [
            "id",
            "type",
            "name",
            "organization",
            "kpiValue",
            "calculationFrequency",
            "category",
            "calculationPeriod"
          ]
        },
        {
          "anyOf": [
            {
              "required": [
                "process"
              ]
            },
            {
              "required": [
                "product"
              ]
            }
          ]
        }
      ]
    }
```
- [UserContext](https://github.com/Fiware/dataModels/blob/master/specs/User/UserContext/schema.json) has `anyOf` at the root level, but with nested and i guess redundant `required` fields:
```
"anyOf": [
    {
      "required": [
        "id",
        "type",
        "location",
        "refUser"
      ]
    },
    {
      "required": [
        "id",
        "type",
        "address",
        "refUser"
      ]
    },
    {
      "required": [
        "id",
        "type",
        "refActivity",
        "refUser"
      ]
    }
  ]
``` 

So, how do we proceed?